app/ide-desktop/lib/dashboard/src/components/AriaComponents/Form/Form.tsx

```diff
@@ -0,0 +1,131 @@
+/**
+ * @file
+ *
+ * Form component
+ */
+import * as React from 'react'
+
+import * as reactQuery from '@tanstack/react-query'
+import * as reactHookForm from 'react-hook-form'
+
+import * as textProvider from '#/providers/TextProvider'
+
+import * as components from './components'
+import type * as types from './types'
+
+/**
+ * Form component. It wraps the form and provides the form context.
+ * It also handles the form submission.
+ * Provides better error handling and form state management.
+ * And serves a better UX out of the box.
+ *
+ * ## Component is in BETA and will be improved in the future.
+ */
+// There is no way to avoid type casting here
+// eslint-disable-next-line no-restricted-syntax
+export const Form = React.forwardRef(function Form<
+  TFieldValues extends reactHookForm.FieldValues,
+  // This type is defined on library level and we can't change it
+  // eslint-disable-next-line no-restricted-syntax
+  TTransformedValues extends reactHookForm.FieldValues | undefined = undefined,
+>(props: types.FormProps<TFieldValues, TTransformedValues>, ref: React.Ref<HTMLFormElement>) {
+  const formId = React.useId()
+
+  const {
+    children,
+    onSubmit,
+    formRef,
+    form,
+    formOptions = {},
+    className,
+    style,
+    onSubmitted = () => {},
+    onSubmitSuccess = () => {},
+    onSubmitFailed = () => {},
+    id = formId,
+    schema,
+    ...formProps
+  } = props
+
+  const { getText } = textProvider.useText()
+
+  const innerForm = components.useForm<TFieldValues, TTransformedValues>(
+    form ?? {
+      ...formOptions,
+      ...(schema ? { schema } : {}),
+    }
+  )
+
+  React.useImperativeHandle(formRef, () => innerForm, [innerForm])
+
+  const formMutation = reactQuery.useMutation({
+    mutationKey: ['FormSubmit', id],
+    mutationFn: async (fieldValues: TFieldValues) => {
+      try {
+        await onSubmit(fieldValues, innerForm)
+      } catch (error) {
+        innerForm.setError('root.submit', {
+          message: error instanceof Error ? error.message : getText('arbitraryFormErrorMessage'),
+        })
+        // TODO: Should we throw the error here?
```

Comment:
TODO without issue reference. but we should probably prefer `toastAndLog` - but on a side note it seems like `react-query` catches and logs thrown errors so there's not much difference between logging and throwing in the first place.

Reply:
I'm not sure it is worth a separate issue. Using toast to display errors isn't a good UX(in this situation), but logging might be helpful... On the other hand, if we log all errors in the console (thus catching them with sentry, we'll end up with a ton of false-positive noise in sentry)

Reply:
in that case we might want to suppress *only* network errors using `is-network-error`. it's true that if there will be a more integrated way of notifying the user that a form submit has failed then that's much better than toasts, *but* we want to make sure that we do tell the user clearly that their form submit failed (and that they will probably want to resubmit.)

Reply:
For showing submit errors, we have a dedicated `<FormError />` component. By calling `innerForm.setError('root.submit',...)` we pass the error to this component.